frontend/src/js/tooltip/Tooltip.tsx

```diff
@@ -164,9 +171,23 @@ const Tooltip = () => {
         />
         <Head>
           <PinnedLabel>
-            <TypeIcon icon={isFolder ? faFolder : faMinus} />
+            <TypeIcon
+              icon={
+                // parent?.label is not element tooltipped or tooltip is a folder
+                isFolder || parent?.label !== label
+                  ? // If the parent is a struct folder, we show the regular folder icon
+                    !parent?.detailsAvailable
+                    ? faFolderRegular
+                    : faFolder
+                  : faMinus
```

Comment:
In order to make this more readable, can I ask you to
- extract this to a function
- avoid nested ternary 
- avoid comments that say the same thing that the code says

Reply:
@Kadrian would this be readable enough for you? I extracted it to a function and added the extracted variables.
```ts
  const chooseIcon = () => {
    const hasParent = parent?.label !== label;
    if(isFolder || hasParent) {
      const isStructNode = !parent?.detailsAvailable;
      if(isStructNode) {
        return faFolderRegular
      }
      return faFolder
    }
    return faMinus
  }
```

Reply:
Hmm, that's a good start. But what's confusing to me there is you'd be saying: "if it does **not** have a parent, return a faMinus" But faMinus concepts usually do have a parent. Maybe `hasParent` is not a good name here? Also: will this benefit from early returns?

Reply:
definitely the second one!

Reply:
Just implemented that one, should hopefully be good now. Check are failing due to java codeql

Reply:
The Java codeql Problem seems to be a spurious Network Glitch durding dependency download. For now we don't have fix for this